src/formatDate.js

```diff
@@ -50,7 +50,61 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let day = 0;
+  let month = 0;
+  let year = 0;
+  const separator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const arrayFromDate = date.split(separator);
+  const newDateArray = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'DD') {
+      day = arrayFromDate[i];
+    };
+
+    if (fromFormat[i] === 'MM') {
+      month = arrayFromDate[i];
+    };
+
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      year = arrayFromDate[i];
+    }
+  };
+
+  for (const element of toFormat) {
+    if (element === 'DD') {
+      newDateArray.push(day);
+    };
+
+    if (element === 'MM') {
+      newDateArray.push(month);
+    };
+
+    if (
```

Comment:
The conditions here can be simplified and the use of multiple 'if' statements can be avoided. Consider using a switch statement or an object as a map for the format.